### PR TITLE
Documents all API endpoints with purpose and function

### PR DESCRIPTION
**Pull request by dennisgove**

### CODE REVIEW

**Comment by jasnell** [comment]
Ha!! Looks like we crossed time streams a bit lol https://github.com/covidgreen/covid-green-backend-api/issues/12

File: lib/routes/callback/index.js
Comment (jasnell) [inline lib/routes/callback/index.js]: @jackmurdoch ... can you confirm the behaviors here?

**Review by jasnell — APPROVED** [review]
LGTM but would like @jackmurdoch to review and sign off before landing

File: lib/routes/callback/index.js
Comment (colmharte) [reply lib/routes/callback/index.js]: The `last_callback` is used to rate limit the callback end point, currently only 1 callback is allowed. This restriction may need to be revised depending on specific needs of each county / state.

File: lib/routes/callback/index.js
Comment (colmharte) [inline lib/routes/callback/index.js]: The `last_callback` is currently not cleared, in Ireland the requirement was one callback only per person. Clearing the last_callback would allow a subsequent callback to be queued for the same person

File: lib/routes/exposures/index.js
Comment (colmharte) [inline lib/routes/exposures/index.js:213]: The limit parameter is no longer supported due to changes in file generation. Multiple files are now generated to allow users to catch up on days they may have missed by downloading a single file.

Comment (dennisgove) [reply lib/routes/exposures/index.js:213]: Thanks @colmharte. I've updated the comments to include this change.

**Review by jasnell — APPROVED** [review]

File: lib/routes/callback/index.js
Comment (dennisgove) [reply lib/routes/callback/index.js]: I've updated the docs to indicate this.

File: lib/routes/callback/index.js
Comment (dennisgove) [reply lib/routes/callback/index.js]: Have updated the docs to indicate this.

File: lib/plugins/verify/index.js
Comment (dennisgove) [inline lib/plugins/verify/index.js:64]: Fixes #14: Documents why this check exists, what it means, and how to disable it.

File: lib/routes/callback/index.js
Comment (ShaunBaker) [inline lib/routes/callback/index.js]: "Temporarily stored"

File: lib/routes/callback/index.js
Comment (ShaunBaker) [inline lib/routes/callback/index.js:15]: Is this an actual issue? Given the limited data stored elsewhere what is personal beyond the mobile number?

File: lib/routes/callback/index.js
Comment (ShaunBaker) [inline lib/routes/callback/index.js]: This is the second IFF I've seen. We should phrase correctly

File: lib/routes/callback/index.js
Comment (ShaunBaker) [inline lib/routes/callback/index.js]: Clarity required here. Implementation can be amended in the app as to whether callback number can be submitted with/out contact happening

File: lib/routes/callback/index.js
Comment (ShaunBaker) [inline lib/routes/callback/index.js:40]: nit: existing and exists in same sentence

File: lib/routes/check-in/index.js
Comment (ShaunBaker) [inline lib/routes/check-in/index.js]: Note the symptom checker in the app is an optional piece of functionality

File: lib/routes/exposures/index.js
Comment (ShaunBaker) [inline lib/routes/exposures/index.js]: This should be 404 no?

File: lib/routes/exposures/index.js
Comment (ShaunBaker) [inline lib/routes/exposures/index.js:128]: @jackmurdoch

File: lib/routes/callback/index.js
Comment (dennisgove) [reply lib/routes/callback/index.js]: I agree. I'll change both to the full "if and only if" for clarity.

File: lib/routes/exposures/index.js
Comment (dennisgove) [reply lib/routes/exposures/index.js]: The current code implementation returns a 403 - Forbidden in 2 cases:

1. The rate limit check/write on table `verifications` fails either because the control doesn't exist or too many requests within a period of time.
1. The code/control cannot be deleted from the `verifications` table.

Whether a 404 is a better result code is up for debate, but the documented 403 is accurate as currently implemented. That said, the reason could be clarified to include the rate limit check.

File: lib/routes/callback/index.js
Comment (dennisgove) [reply lib/routes/callback/index.js]: This paragraph was written based on what was in Nearform's App repository. Looking at https://github.com/covidgreen/covid-green-app it doesn't appear this endpoint is in the default base App. I can either remove the paragraph or rephrase it along the lines of "one possible implementation / integration with the App could be ..."

What do you think?

File: lib/routes/exposures/index.js
Comment (dennisgove) [reply lib/routes/exposures/index.js]: Added additional clarity in latest commit.

File: lib/routes/check-in/index.js
Comment (dennisgove) [reply lib/routes/check-in/index.js]: Removed mention of what the App allows a user to do and focuses on what this endpoint supports.

File: lib/routes/callback/index.js
Comment (dennisgove) [reply lib/routes/callback/index.js:40]: Fixed in latest commit.

File: lib/routes/callback/index.js
Comment (dennisgove) [reply lib/routes/callback/index.js]: Latest commit removes mention of what the app allows or supports.

File: lib/routes/callback/index.js
Comment (dennisgove) [reply lib/routes/callback/index.js:15]: I think the potential for a PII/GDPR issue exists and as such should be documented. As currently implemented, this repository loses control over what happens with the User ID and phone# once they're added to the AWS queue.

The latest commit rephrases this entire paragraph to hopefully clarify the issue.

Comment (dennisgove) [reply lib/routes/callback/index.js:15]: I do have a potential solution that closes this PII/GDPR issue completely (and allows for additional features like cancelling callbacks and updating phone #s). I want to give it a little more thought before I document it, but do want to share that I've got something brewing.

Comment (dennisgove) [reply lib/routes/callback/index.js:15]: I've entered #20 to discuss a potential solution.

**Comment by dennisgove** [comment]
I believe all the review comments/suggestions have been dealt with. It still needs 1 final approval before merging.

**Review by fiacc — APPROVED** [review]
👍

**Pull request merged**
